float32 x
float32 y
float32 z
float32 phi
float32 theta
float32 psi
---
float32 q1
float32 q2
float32 q3
